printer_zpl2/lib/zpl2.py

```diff
@@ -0,0 +1,335 @@
+
+        # Check that the height value fits in the allowed values
+        if fontFormat.get('height') is not None:
+            fontFormat['height'] = self._enforce(
+                fontFormat['height'], minimum=10)
+
+        # Check that the width value fits in the allowed values
+        if fontFormat.get('width') is not None:
+            fontFormat['width'] = self._enforce(
+                fontFormat['width'], minimum=10)
+
+        # Generate the ZPL II command
+        return '^A' + self._generateArguments(arguments, fontFormat)
+
+    def _fieldBlock(self, blockFormat):
+        """ Define a maximum width to print some data """
+        arguments = [
+            'block_width',
+            'block_lines',
+            'block_spaces',
+            'block_justify',
+            'block_left_margin',
+        ]
+        return '^FB' + self._generateArguments(arguments, blockFormat)
+
+    def _barcodeFormat(self, barcodeType, barcodeFormat):
+        """ Generate the commands to print a barcode
+        Each barcode type needs a specific function
+        """
+        def _code11(**kwargs):
+            arguments = [
+                'orientation',
+                'check_digit',
+                'height',
+                'interpretation_line',
+                'interpretation_line_above',
+            ]
+            return '1' + self._generateArguments(arguments, kwargs)
+
+        def _interleaved2of5(**kwargs):
+            arguments = [
+                'orientation',
+                'height',
+                'interpretation_line',
+                'interpretation_line_above',
+                'check_digit',
+            ]
+            return '2' + self._generateArguments(arguments, kwargs)
+
+        def _code39(**kwargs):
+            arguments = [
+                'orientation',
+                'check_digit',
+                'height',
+                'interpretation_line',
+                'interpretation_line_above',
+            ]
+            return '3' + self._generateArguments(arguments, kwargs)
+
+        def _code49(**kwargs):
+            arguments = [
+                'orientation',
+                'height',
+                'interpretation_line',
+                'starting_mode',
+            ]
+            # Use interpretation_line and interpretation_line_above to generate
+            # a specific interpretation_line value
+            if kwargs.get('interpretation_line') is not None:
+                if not kwargs['interpretation_line']:
+                    kwargs['interpretation_line'] = 'N'
+                if kwargs['interpretation_line'] and not kwargs[
+                   'interpretation_line_above']:
+                    kwargs['interpretation_line'] = 'B'
+                else:
+                    kwargs['interpretation_line'] = 'A'
+            return '4' + self._generateArguments(arguments, kwargs)
+
+        def _pdf417(**kwargs):
+            arguments = [
+                'orientation',
```

Comment:
These correspond to Odoo fields names, so I think this can (should ?) be constants.
I'll refactor that when moving the lib to a separate python package.